src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -3465,7 +3474,25 @@ namespace SimAirServingZones {
             } else if (SELECT_CASE_var == CoilUserDefined) { // Coil:UserDefined
                 SimCoilUserDefined(CompName, CompIndex, AirLoopNum, HeatingActive, CoolingActive);
 
-            } else if (SELECT_CASE_var == UnitarySystem) { // 'AirLoopHVAC:UnitarySystem'
+            } else if (SELECT_CASE_var == UnitarySystemModel) { // 'UnitarySystem'
+                // set up a reference for this component
+                bool foundComp = false;
+                for (int branchNum = 1; branchNum <= PrimaryAirSystem(AirLoopNum).NumBranches; ++branchNum) {
+                    for (int compNum = 1; compNum <= PrimaryAirSystem(AirLoopNum).Branch(branchNum).TotalComponents; ++compNum) {
+                        if (CompName == PrimaryAirSystem(AirLoopNum).Branch(branchNum).Comp(compNum).Name) {
+                            auto &sim_component(PrimaryAirSystem(AirLoopNum).Branch(branchNum).Comp(compNum));
+                            int OAUnitNum = 0;
+                            Real64 OAUCoilOutTemp = 0.0;
+                            bool ZoneEquipFlag = false;
+                            sim_component.compPointer->simulate(CompName, FirstHVACIteration, AirLoopNum, CompIndex, HeatingActive, CoolingActive, OAUnitNum, OAUCoilOutTemp, ZoneEquipFlag);
```

Comment:
@Myoldmopar I am not doing this right. When I get to the simulate routine "this->" isn't pointing to the correct system. It's as if it is pointing to a new system. In fact getInput is called again and now unitarySys.size() = 3 when there are only 2 systems in the input file. This is at line 3487 of SimAirServingZones. The factory is called on line 1297.

I tried this method in the factory with the same result. Any suggestions?

    int sysNum = -1;
    for (auto &sys : unitarySys) {
        ++sysNum;
        if (UtilityRoutines::SameString(sys.name, objectName) && sys.unitarySystemType_Num == object_type_of_num) {
            if (unitarySys[sysNum].myGetInputSuccessfulFlag) unitarySys[sysNum].thisSysInputShouldBeGotten = false;
            unitarySys[sysNum].unitarySysNum = sysNum;
            return &sys;
        }
    }